### PR TITLE
Importing SSH Keys

### PR DESCRIPTION
**Pull request by javierdemartin**
Tapping on private and public SSH keys on Files.app and other apps imports those keys inside `.ssh/` folder.

### CODE REVIEW

**Comment by yury** [comment]
I think, the idea of this issue it import private key to our Secure Storage not copy to file system. You should add another option to `config`->`keys`->`+`. It should behave almost the same as `Import from clipboard` but use FileBrowser for file to import

**Comment by javierdemartin** [comment]
Got it, will re-do it

File: Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m [hunk 1]
```diff
@@ -299,4 +349,30 @@ - (void)showKeyInfo:(NSIndexPath *)indexPath
...
+#pragma mark - UIDocumentPickerDelegate
+
+- (void)documentPickerWasCancelled:(UIDocumentPickerViewController *)controller {
+  
+  
+}
+
+- (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls {
+  
+  NSArray *readFileUrls = urls;
```
Comment (yury) [inline Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m:361]: Why we need this var?

File: Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m [hunk 2]
```diff
@@ -169,46 +242,23 @@ - (void)createKey {
...
+  picker.allowsMultipleSelection = true;
+  picker.delegate = self;
+  
+  [self presentViewController:picker animated:true completion:nil];
+  
+}
+
 - (void)importKey
 {
   UIPasteboard *pb = [UIPasteboard generalPasteboard];
```
Comment (yury) [inline Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m:257]: Don't think we need that var anymore :)

File: Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m [hunk 3]
```diff
@@ -158,6 +230,7 @@ - (IBAction)addKey:(id)sender
 
   [keySourceController addAction:generate];
   [keySourceController addAction:import];
```
Comment (yury) [inline Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m:232]: We could rename `import` to `importFromClibpoard` here.

File: Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m [hunk 4]
```diff
@@ -299,4 +349,30 @@ - (void)showKeyInfo:(NSIndexPath *)indexPath
...
+  
+}
+
+- (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls {
+  
+  NSArray *readFileUrls = urls;
+  
+  for (NSURL *fileUrl in readFileUrls) {
+    
+    NSString *keyString = [NSString stringWithContentsOfURL:fileUrl encoding:NSUTF8StringEncoding error:NULL];
```
Comment (yury) [inline Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m:365]: We can call `documentPicker:didPickDocumentAtURL:` here to avoid duplication.

Comment (javierdemartin) [reply Settings/ViewControllers/BKPubKey/BKPubKeyViewController.m:365]: Removed implementation of `documentPicker:didPickDocumentAtURL:` it's deprecated. We're only gonna use `documentPicker:didPickDocumentsAtURLs:`.

**Pull request merged**